scripts/o3de/o3de/register.py

```diff
@@ -472,13 +472,15 @@ def register_restricted_path(json_data: dict,
 def register_repo(json_data: dict,
                   repo_uri: str,
                   remove: bool = False) -> int:
-    if not repo_uri:
-        logger.error(f'Repo URI cannot be empty.')
+    manifest_uri = repo.get_repo_manifest_uri(repo_uri)
```

Comment:
```get_repo_manifest_uri``` doesn't seem to add too much value here other. Maybe we can update it to do the following:
1. Form the URI (with the /repo.json path)
2. Perform the urllib parse in it as well so you can validate it to be non-empty and a valid url
3. Update the signature to return a tuple ( manifest_uri, parsed_uri )